openhands/core/config/mcp_config.py

```diff
@@ -32,6 +33,21 @@ class MCPStdioServerConfig(BaseModel):
     args: list[str] = Field(default_factory=list)
     env: dict[str, str] = Field(default_factory=dict)
 
+    def __eq__(self, other):
+        """Override equality operator to compare server configurations.
+
+        Two server configurations are considered equal if they have the same
+        name, command, args, and env values.
+        """
+        if not isinstance(other, MCPStdioServerConfig):
+            return False
+        return (
+            self.name == other.name
+            and self.command == other.command
+            and self.args == other.args
+            and self.env == other.env
+        )
+
```

Comment:
Would this work if two server have the same args but in different orders?

Reply:
Maybe we could also write a small test to confirm the expected equality behavior

Reply:
@OpenHands please convert them to set before comparison. Then add a unit test in tests/unit/test_mcp_config.py to validate that